### PR TITLE
bgpv1: add CiliumBGPFamily CRD and utilize it

### PR DESCRIPTION
**Pull request by ldelossa**
This commit introduces a new CiliumBGPFamily CRD which allows users to specify which Safi/Afi(s) they'd like to advertise or learn over a peer.

If no CiliumBGPFamily(s) are provided in the CiliumBGPNeighbor CRD, a default set is provided which advertise/learn IPv4 and IPv6 prefixes.

```release-note
Introduce ability to specify SAFI/AFI for specific BGP peers. 
```

### CODE REVIEW

**Review by rastislavs — APPROVED** [review]

File: pkg/bgpv1/gobgp/server.go [hunk 1]
```diff
@@ -177,13 +223,13 @@ func (g *GoBGPServer) getPeerConfig(ctx context.Context, n types.NeighborRequest
...
 			Conf:      existingPeer.Conf,
 			Transport: existingPeer.Transport,
-			AfiSafis:  existingPeer.AfiSafis,
 		}
 		// Update the peer port if needed.
 		if existingPeer.Transport.RemotePort != peerPort {
 			peer.Transport.RemotePort = peerPort
 		}
 	} else {
+
```
Comment (tommyp1ckles) [inline pkg/bgpv1/gobgp/server.go]: nit: unecessary newline

File: pkg/bgpv1/gobgp/server.go [hunk 2]
```diff
@@ -193,23 +239,14 @@ func (g *GoBGPServer) getPeerConfig(ctx context.Context, n types.NeighborRequest
...
-						Family: GoBGPIPv6Family,
-					},
-				},
-			},
 		}
 	}
 
+	peer.AfiSafis, err = convertBGPNeighborSAFI(n.Neighbor.Families)
+	if err != nil {
+		return peer, needsReset, fmt.Errorf("failed to convert CiliumBGPNeighbor Families to gobgp AfiSafi: %w", err)
```
Comment (tommyp1ckles) [inline pkg/bgpv1/gobgp/server.go:246]: nit: might be helpful to add the offending afi/safi to this error (or the "Unknown X" one?) to make it clear to users what field failed.

**Review by tommyp1ckles — APPROVED** [review]
k8s changes lgtm

File: pkg/bgpv1/gobgp/server.go (hunk 2 above)
Comment (ldelossa) [reply pkg/bgpv1/gobgp/server.go:246]: cool, I made this happen by way of error wrapping. 

```
func (a *Afi) FromString(s string) error {
	switch s {
	case "ipv4":
		*a = AfiIPv4
	case "ipv6":
		*a = AfiIPv6
	case "l2vpn":
		*a = AfiL2VPN
	case "ls":
		*a = AfiLS
	case "opaque":
		*a = AfiOpaque
	default:
		return fmt.Errorf("Unknown Afi: %s", s)
	}
	return nil
}
```

these errors will be wrapped all the way up to the line you're pointing out here.

**Comment by joestringer** [comment]
Is this initial work towards MP-BGP? I wonder whether this or subsequent work in this direction may be appropriate for `release-note/major` as a candidate for v1.15 release blog to talk about extending Cilium's BGP support.

**Comment by rastislavs** [comment]
@joestringer 
> Is this initial work towards MP-BGP? I wonder whether this or subsequent work in this direction may be appropriate for `release-note/major` as a candidate for v1.15 release blog to talk about extending Cilium's BGP support.

This does not really introduce any new functionality (yet), just makes the configuration a bit more more explicit for future extensions. That said, IMO `minor` fits well here.

**Pull request merged**